src/core/lib/event_engine/tcp_socket_utils.cc

```diff
@@ -70,6 +72,45 @@ absl::StatusOr<std::string> GetScheme(
           "Unknown scheme: %d", resolved_address.address()->sa_family));
   }
 }
+
+#ifdef GRPC_HAVE_UNIX_SOCKET
+absl::StatusOr<std::string> ResolvedAddrToUriUnixIfPossible(
+    EventEngine::ResolvedAddress* resolved_addr) {
+  const sockaddr* addr = resolved_addr->address();
+  if (addr->sa_family != AF_UNIX) {
+    return absl::InvalidArgumentError(
+        absl::StrCat("Socket family is not AF_UNIX: ", addr->sa_family));
+  }
+  const sockaddr_un* unix_addr = reinterpret_cast<const sockaddr_un*>(addr);
+  bool abstract =
+      resolved_addr->size() < sizeof(unix_addr->sun_family) + 1 ||
+      (unix_addr->sun_path[0] == '\0' && unix_addr->sun_path[1] != '\0');
+  std::string scheme;
+  std::string path;
+  if (abstract) {
+    scheme = "unix-abstract";
+    if (resolved_addr->size() >= sizeof(unix_addr->sun_family) + 1) {
```

Comment:
So it is possible for `path` to be an empty string?

Reply:
Yes. The path can be empty. The purpose of this function is to merely return the path no matter what the path maybe.